### ISSUE TITLE
Add an `fc.double`/`fc.float` variant that strictly produces non-integers 

### ISSUE THREAD

**Issue by TomerAberbach** [dubzzz/fast-check#4831]
## 🚀 Feature Request and Motivation

Sometimes I want to test something about my function for specific types of numbers. If I want to test any number, any number in some range, any integer, any integer in some range, then I can do that using the built in arbitraries.

However, something I can't easily do is test non-integers specifically. `fc.double` and `fc.float` _can_ generate integers. Understood that all JS numbers are floating point numbers, but I basically mean I can't specifically test numbers that do not pass `Number.isInteger`.

So far I have been doing `fc.double(...).filter(value => !Number.isInteger(value))`, but I'm guessing this isn't as efficient as it could be. Perhaps if this were built into `fast-check`, then there could be a faster implementation (and also I wouldn't have to write it over and over again).

## Example

Probably something like this:

```js
fc.double({ noInteger: true })
fc.float({ noInteger: true })
```

Seems like that API would make sense since the param doesn't affect the return type (still `number`) and there are other similarly named params in the API (`noDefaultInfinity`, `noNaN`, etc.)

**Comment by dubzzz** [comment]
Why not 🤔 It seems legit. We would probably have to do some tweaks as all very huge doubles are integers so we could probably forbid ranges known to only be integers. Same at generate time, noInteger implicitly cap the range as passed a given value we reach integer only world (large but integers).

**Comment by dubzzz** [comment]
Ending a few PRs aiming to make fast-check ESM by default and then I may take that one

**Comment by TomerAberbach** [comment]
Nice, thanks!

**Comment by dubzzz** [comment]
Currently POC-ing an implementation locally. I'm wondering if I'd also need to offer a noFloat. In such case either I offer noInteger and noFloat or just a kind:float/integer/both. Then there is the question: is nan a float? is infinity a float?.. they should probably... so noFloat would conflict with other flags 🤯

**Comment by dubzzz** [comment]
Just to confirm your idea of noInteger: by default it would return things -inf, nan, +inf and 4503599627370495.5 (the max float) plus any float.  Am I right?

**Comment by dubzzz** [comment]
Out of curiosity, in which case do you need such thing?

**Comment by TomerAberbach** [comment]
> Currently POC-ing an implementation locally. I'm wondering if I'd also need to offer a noFloat. In such case either I offer noInteger and noFloat or just a kind:float/integer/both. Then there is the question: is nan a float? is infinity a float?.. they should probably... so noFloat would conflict with other flags 🤯
> Just to confirm your idea of noInteger: by default it would return things -inf, nan, +inf and 4503599627370495.5 (the max float) plus any float. Am I right?

I was generally thinking that `noInteger: true` would basically give you the same result as `fc.double(...).filter(value => !Number.isInteger(value))` and `fc.float(...).filter(value => !Number.isInteger(value))`. Open to counter arguments about specific values, but I think `Number.isInteger` is pretty authoritative

> Out of curiosity, in which case do you need such thing?

Sometimes a function allows `number` as the input type, but disallows non-integers (i.e. it throws for them) or it has different behavior for integers vs doubles/floats. In this case `fc.double({ noInteger: true })` would help testing the doubles/floats case in its own test. Hope that makes sense!

**Comment by dubzzz** [comment]
I have a POC ready for double, I just need to play a bit more with it

**Comment by dubzzz** [comment]
I will soon merge a first iteration around it for double: https://github.com/dubzzz/fast-check/pull/4917

Then I need to address the few points still opened (see description of the PR) and I'll release it!

**Comment by dubzzz** [comment]
Released at 3.18.0. Feel free to re-open in case it does not meet the expectations you had when opening the request.

**Comment by TomerAberbach** [comment]
Works like a charm, thanks!
